packages/base/README.md

```diff
@@ -1,2 +1,94 @@
-# appium-base-plugin
-The base class used to create Appium 2.0 plugins
+# Appium Base Plugin
+
+The base class used to create Appium plugins. This plugin should not be installed directly as it does nothing. Instead, you extend this plugin when creating your *own* Appium plugins.
+
+## Steps to create your own plugin
+
+You can follow this guide to create your own Appium plugin.
+
+### 1. Initialize a Node.js project
+
+Appium plugins are Node.js programs, though of course you can call out to any other language or framework you want in the course of executing Node.js code. The basic requirement for an Appium plugin is to add certain metadata fields in your project's `package.json`, for example:
+
+```
+{
+  ...,
+  "appium": {
+    "pluginName": "your-plugin-name",
+    "mainClass": "YourPlugin",
+  },
+  ...
+}
+```
+
+Here the `pluginName` is the name used to activate your plugin by users who have installed it. So if your `pluginName` is `cool-plugin`, they would activate it as follows:
+
+```
+appium --plugins=cool-plugin
+```
+
+The same goes for updating and removing the plugin using the CLI.
+
+The `mainClass` value is the name that Appium should import from your project's entrypoint. So for example if somewhere in your project you have defined your plugin as follows:
```

Comment:
> Do we expect plugin authors to export multiple classes? If not we can just expect an ESM default export and can avoid having to define this.

Plugins might export other things, or one package might export both a driver and a plugin, for example. I thought it best to require an explicit export, but we could also have a convention where if `mainClass` is not present, then the default export is used.

Reply:
> Plugins could be used as middlewares, e.g. before a command is send (to modify the payload) or after response has been received (to modify the response). Appium can define all these types and then expect plugin authors to have respective named exports for them.

Yes, this is a good idea. However, in that case, the various exported plugin behaviors would not cohere as easily with one another. I thought a class-based approach was interesting because one instance of a plugin could have state shared between all its expressions (server updater, command handler, etc...)